app/services/group_aware_role_checker.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Loop group memberships into the role-checking process
+module GroupAwareRoleChecker
+  # Dynamically define all #<role_name>? methods so that, as more roles are added,
+  # their role checker methods are automatically defined
+  RolesService::DEFAULT_ROLES.each do |role_name|
+    define_method(:"#{role_name}?") do
+      has_group_aware_role?(role_name)
+    end
+  end
+
+  private
+
+    # Check for the presence of the passed role_name in the User's Roles and
+    # the User's Hyrax::Group's Roles.
+    # TODO: investigate why this method causes spec/features/appearance_theme_spec.rb to fail
+    def has_group_aware_role?(role_name)
+      return false if current_user.new_record?
+
+      current_user.reload # Get the most up-to-date version of the User before checking for Role
```

Comment:
reload is an extra db call. that tends to be expensive performance wise. is it necessary here?

Reply:
@bkiahstroud also - is this your spec bug?!?

Reply:
I _think_ it was intended to be a cache-buster; the ability stuff gets cached relatively heavily (since they don't change often, that makes sense), but this is a crucial method when it comes to deciding who can do what, so I think I just wanted it to be as up-to-date possible. 

I'll remove it and see what impact that has.